### ISSUE TITLE
Multi-span messages are not matched

### ISSUE THREAD

**Issue by pczarn** [laumann/compiletest-rs#41]
When multi-span messages are emitted, the compiler duplicates the message for each span in the JSON output. Compiletest recognizes the first message, and reports mismatch for the rest.

**Comment by laumann** [comment]
Sorry for the late reply. Is this also an issue in rustc's compiletest?

**Comment by pczarn** [comment]
Yes, the rustc issue is https://github.com/rust-lang/rust/issues/31380
